charts/vela-core/templates/defwithtemplate/gateway.yaml

```
@@ -44,6 +44,16 @@ spec:
         		if parameter.classInSpec {
         			ingressClassName: parameter.class
         		}
+        		if parameter.tlsInSpec {
```

what about use `secretName` to check if we use tls or not, then we don't need a redundant `tlsInSpec` field.

```suggestion
        		if parameter.secretName != _|_ {
```